src/lib.rs

```diff
@@ -673,25 +887,31 @@ impl<T> core::future::Future for Receiver<T> {
 
 impl<T> Drop for Receiver<T> {
     fn drop(&mut self) {
-        // SAFETY: The reference won't be used after it is freed in this method
+        // SAFETY: since the receiving side is still alive the sender would have observed that and
+        // left deallocating the channel allocation to us.
         let channel = unsafe { self.channel_ptr.as_ref() };
 
         // Set the channel state to disconnected and read what state the receiver was in
-        match channel.state.swap(DISCONNECTED, SeqCst) {
+        match channel.state.swap(DISCONNECTED, Acquire) {
             // The sender has not sent anything, nor is it dropped.
             EMPTY => (),
             // The sender already sent something. We must drop it, and free the channel.
             MESSAGE => {
+                // SAFETY: we are in the message state so the message is initialized
                 unsafe { channel.drop_message() };
+
+                // SAFETY: see safety comment at top of function
                 unsafe { dealloc(self.channel_ptr) };
             }
             // The receiver has been polled.
             #[cfg(feature = "async")]
             RECEIVING => {
+                // TODO: figure this out when async is fixed
```

Comment:
What's the background of TODO here? I don't find it's still valid now.

Reply:
I'm guessing this was a note to think about what would need to happen if the receiver was dropped while in this state. In the sync version this is not possible, since either the receiver is owned or guaranteed to remain alive because of a borrow, but in an async context this is technically possible, although I think the situations in which such a thing would happen and be observable by a program are fairly pathological. It's entirely possible that this is the correct thing to do (and at a glance I'd believe it), but it could use some more careful thought, perhaps.